libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -586,18 +590,17 @@ uint8_t AC_Fence::check(bool armed)
         ret |= AC_FENCE_TYPE_ALT_MAX;
     }
 
-    // auto enable floor check
-    if (!armed && option_enabled(AC_Fence::OPTIONS::AUTO_DISABLE_FLOOR_ON_DISARM)) {
-        disable_floor();    // this also resets the alt min fence breach
-    } else if (!_floor_enabled && auto_enabled() == AC_Fence::AutoEnable::ALWAYS_ENABLED) {
-        auto_enable_fence_floor();
-    }
-
-    // minimum altitude fence check
+    // minimum altitude fence check, do this before auto-disabling (e.g. because falling)
+    // so that any action can be takem
     if (_floor_enabled && check_fence_alt_min()) {
         ret |= AC_FENCE_TYPE_ALT_MIN;
     }
 
+    // auto enable floor check
+    if (auto_enabled() == AC_Fence::AutoEnable::ALWAYS_ENABLED) {
```

Comment:
why not universally? if no MIN_ALT setup you quick return.....this enables MIN_ALT once you get above it all the time....floor starts off always disabled even though setup fences are (if FENCE_ENABLE=1 at boot) unless an overt fence enable is done

Reply:
Fixed thanks